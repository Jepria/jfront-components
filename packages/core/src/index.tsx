export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
