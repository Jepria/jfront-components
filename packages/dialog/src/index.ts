export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'