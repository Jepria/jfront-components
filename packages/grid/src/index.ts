export * from './common'
export * from './jepgrid'
