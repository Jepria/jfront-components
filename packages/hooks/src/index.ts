export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
