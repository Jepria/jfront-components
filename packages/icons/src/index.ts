export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
