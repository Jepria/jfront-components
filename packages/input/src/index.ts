export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
