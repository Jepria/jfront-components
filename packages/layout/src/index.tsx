import styled from 'styled-components';

export const FormContainer = styled.div`
  padding: 5px;
`;

export const VerticalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const HorizontalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
`;

export const Page = styled.div`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const Header = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;

export const Content = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -ms-flex-positive: 1;
      flex-grow: 1;
`;

export const Footer = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;
