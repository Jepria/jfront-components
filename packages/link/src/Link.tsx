import React from "react"
import { forwardRef } from "@jfront/ui-utils"
import styled from "styled-components"

const StyledSpan = styled.span`
  color: currentColor;
  font-weight: bold;
`

const StyledLink = styled.a`
  color: currentColor;
  text-decoration: none;
  font-weight: normal;
  &:hover {
    opacity: 0.7;
    text-decoration: underline;
  }
`

export interface LinkProps
  extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
  method?: string //"redirect" || "history"
  title?: string
  children: React.ReactNode
}

export const Link = forwardRef<LinkProps, "a">((props, ref) => {
  const { method = "history", href, title = "", as, onClick, ...rest } = props

  const navigate = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
    e.preventDefault()
    e.stopPropagation()
    if (onClick != undefined) {
      onClick(e)
    }
    if (href != undefined) {
      history.replaceState(null, title, href)
    }
  }

  if (as) {
    const Component = as
    return <Component ref={ref} {...props} />
  } else if (href != undefined) {
    return (
      <StyledLink
        {...rest}
        href={method === "history" ? "" : href}
        ref={ref}
        onClick={method === "history" ? navigate : undefined}
      >
        {rest.children}
      </StyledLink>
    )
  } else {
    return (
      <StyledSpan
        {...rest}
        ref={ref}
        onClick={method === "history" ? navigate : undefined}
      >
        {rest.children}
      </StyledSpan>
    )
  }
})
