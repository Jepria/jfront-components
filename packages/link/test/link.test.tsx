import React from "react"
import { render } from "react-dom"
import { Link } from "../src"
import { fireEvent, screen } from "@testing-library/react"

const container = document.createElement("div")
document.body.appendChild(container)

test.skip("Matches snapshot ", () => {
  render(<Link href="/test">Test Link</Link>, container)
  expect(document.body).toMatchSnapshot()
})

test("link click", () => {
  render(
    <Link data-testid="a" href="/test">
      Test Link
    </Link>,
    container,
  )
  fireEvent.click(screen.getByTestId("a"))
  expect(window.location.pathname).toEqual("/test")
})
