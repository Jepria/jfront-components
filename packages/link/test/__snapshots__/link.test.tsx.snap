// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot  1`] = `
<body>
  <div>
    <a
      class="sc-AxirZ fsqDyf"
      href=""
    >
      Test Link
    </a>
  </div>
</body>
`;
